From c7c2b16541cc27b8705dfb925310a1d1811ccb5b Mon Sep 17 00:00:00 2001
From: Akhil Goyal <gakhil@marvell.com>
Date: Thu, 12 Sep 2024 20:59:27 +0530
Subject: [PATCH 502/513] examples/l2fwd-macsec: enable extended packet number

Enabled the extended packet number(XPN) case so that
traffic runs for longer duration without raising interrupt
for PN threshold as the application currently does not handle
rekeying.

Signed-off-by: Akhil Goyal <gakhil@marvell.com>
Change-Id: Id2f3d4f367a4a79c40cff5c486a77bcad3db6e6c
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/135583
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Jerin Jacob <jerinj@marvell.com>
(cherry picked from commit e93e72733f4371a67425f13cd35e8e3844060089)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/135633
Base-Tests: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
---
 examples/l2fwd-macsec/main.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/examples/l2fwd-macsec/main.c b/examples/l2fwd-macsec/main.c
index a5d7a84828034..8c7ec705c1878 100644
--- a/examples/l2fwd-macsec/main.c
+++ b/examples/l2fwd-macsec/main.c
@@ -446,7 +446,8 @@ fill_macsec_sc_conf(uint16_t portid, struct rte_security_macsec_sc *sc_conf)
 		}
 		sc_conf->sc_tx.active = 1;
 		sc_conf->sc_tx.sci = mcs_port_params[portid].sci;
-		if (mcs_port_params[portid].xpn > 0)
+		if (mcs_port_params[portid].alg == RTE_SECURITY_MACSEC_ALG_GCM_XPN_128 ||
+			       mcs_port_params[portid].alg == RTE_SECURITY_MACSEC_ALG_GCM_XPN_256)
 			sc_conf->sc_tx.is_xpn = 1;
 	} else {
 		for (i = 0; i < RTE_SECURITY_MACSEC_NUM_AN; i++) {
@@ -456,7 +457,8 @@ fill_macsec_sc_conf(uint16_t portid, struct rte_security_macsec_sc *sc_conf)
 			}
 		}
 		sc_conf->sc_rx.active = 1;
-		if (mcs_port_params[portid].xpn > 0)
+		if (mcs_port_params[portid].alg == RTE_SECURITY_MACSEC_ALG_GCM_XPN_128 ||
+			       mcs_port_params[portid].alg == RTE_SECURITY_MACSEC_ALG_GCM_XPN_256)
 			sc_conf->sc_rx.is_xpn = 1;
 	}
 }
@@ -1008,7 +1010,7 @@ l2fwd_macsec_default_options(struct l2fwd_macsec_options *options)
 		if ((options->rx_portmask & (1 << portid)) != 0)
 			mcs_port_params[portid].dir = RTE_SECURITY_MACSEC_DIR_RX;
 
-		mcs_port_params[portid].alg = RTE_SECURITY_MACSEC_ALG_GCM_128;
+		mcs_port_params[portid].alg = RTE_SECURITY_MACSEC_ALG_GCM_XPN_128;
 		memcpy(mcs_port_params[portid].sa_key.data, key, 16);
 		mcs_port_params[portid].sa_key.len = 16;
 		memcpy(mcs_port_params[portid].salt, salt, MCS_SALT_LEN);
-- 
2.25.1

